### PR TITLE
[MacOS] fix python installation order

### PR DESCRIPTION
**Pull request by mikhailkoliada**
# Description

The `azure-cli` package pulls python3.9 as a dependency from within the `powershell.sh` installer, which is being called before the actual python installer.  In our `python.sh` script we are installing both python2 from the upstream `.pkg` file and python3 from brew. python2 installation overrides the previously created `/usr/local/bin/2to3` symlink to its own internal path -`/Library/Frameworks/Python.framework/Versions/2.7/bin/2to3`, in the meantime python3 installation attempt is being ignored (as well as the `brew link --overwrite python@3.` call) as it had already been installed by azure cli. To prevent this azure-cli installation has been moved from `powershell.sh` to `commonutils.sh` to fix the installation order.

Apart from that:

- A test has been written to make sure that `2to3` symlink points to the correct place
- `az -v` and the `~/.azure` directory owner fix have been removed (it was figured that they are not needed anymore)

Interesting side effect:

 - Whenever you call `az -v` and then `pwsh -command "& {Import-Module Az}"` the `~/.azure` directory will be created and files created after `Import-Module Az` will be placed there as well
 
This works both ways:

- Whenever you call `pwsh -command "& {Import-Module Az}"` the `~/.Azure` directory will be created and files created after `az -v` call will be placed there.

## Check list
- [X] Related issue / work item is attached
- [X] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [X] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/provision/core/powershell.sh [hunk 1]
```diff
@@ -31,7 +25,4 @@ pwsh -command "& {Import-Module Az}"
 # powershell link was removed in powershell-6.0.0-beta9
 sudo ln -s /usr/local/bin/pwsh /usr/local/bin/powershell
 
-# fix ~/.azure directory permissions
-sudo chown -R ${USER}: $HOME/.azure
```
Comment (maxim-lobanov) [inline images/macos/provision/core/powershell.sh:35]: Looks like we drop this line? Are you sure that it is safe and won't break users?

Comment (mikhailkoliada) [reply images/macos/provision/core/powershell.sh:35]: @maxim-lobanov Yes, we are ok here, I tested it and the directory and the files in it are created with the standard user's permissions and ownership, might be the fix was needed before, but now it is ok per se.

**Review by miketimofeev — APPROVED** [review]

**Review by al-cheb — APPROVED** [review]

**Pull request merged**